pkg/protocol/engine/eviction/state.go

```diff
@@ -215,15 +217,19 @@ func (s *State) LatestRootBlocks() iotago.BlockIDs {
 }
 
 // Export exports the root blocks to the given writer.
-func (s *State) Export(writer io.WriteSeeker, targetSlot iotago.SlotIndex) (err error) {
+func (s *State) Export(writer io.WriteSeeker, lowerTarget iotago.SlotIndex, targetSlot iotago.SlotIndex) (err error) {
```

Comment:
`lowerTarget` means what?

Reply:
Rootblocks are special when creating a snapshot because we not only need them to depend on the slot we're targeting to export (usually last committed slot) but also to derive the rootcommitment. The rootcommitment,  however, **must not** depend on the committed slot but on the finalized slot. Otherwise we could never switch a chain after committing (as the rootcommitment is our genesis and we don't solidify/switch chains below it).